src/base/bittorrent/dbresumedatastorage.cpp

```diff
@@ -854,7 +854,7 @@ namespace
             query.bindValue(DB_COLUMN_NAME.placeholder, m_resumeData.name);
             query.bindValue(DB_COLUMN_CATEGORY.placeholder, m_resumeData.category);
             query.bindValue(DB_COLUMN_TAGS.placeholder, (m_resumeData.tags.isEmpty()
-                    ? QString() : m_resumeData.tags.join(u","_s)));
+                    ? QString() : QStringList(m_resumeData.tags.cbegin(), m_resumeData.tags.cend()).join(u","_s)));
```

Comment:
> I would like to patch `OrderedSet::join()` (by myself) to make the previous code work and avoid the performance regression.

I intentionally did not do something like this, since `join` action is outside the scope of such a generic container. No wonder Qt has an explicit `QStringList` class that extends the base `QList<QString>` container with such QString-specific methods. Now I have come to the conclusion that this is the best way.

Reply:
But anyway `join` is essentially out of terms of `Tag` class. It becomes to strings. So I'd prefer to see explicit sequence of conversions `tags->strings-≥joined_strings`.

Reply:
> Lets face it, `join` has frequent use and it may be operating on a large dataset. I would like to have efficient implementation for it. 

Are you not satisfied with the performance of `QStringList::join()`?
Otherwise, all that remains is to create `QStringList` from the elements of the `TagSet`. Do you think this does at least some serious damage to performance (considering that the `Tag` is essentially an "implicitly shared" class, so it does not lead to copying large amounts of data).
>I already said that I'll going to do it later by myself.

As you wish.
I just want to mention once again that it is incorrect to call "tag set" to "string" transformation "join".

Reply:
>Are you not satisfied with the performance of QStringList::join()?
Otherwise, all that remains is to create QStringList from the elements of the TagSet. Do you think this does at least some serious damage to performance (considering that the Tag is essentially an "implicitly shared" class, so it does not lead to copying large amounts of data).

In short, it wastes CPU cycles. I'm not satisfied with: 1. the usage of QStringList(Iter begin, Iter end) which will loop over the dataset and store the same data in memory (it still takes some memory even if it is implicitly shared). 2. `QStringList::join()` which will loop over the dataset again.

>I just want to mention once again that it is incorrect to call "tag set" to "string" transformation "join".

I think it can be done based on the fact that `Tag` is convertible to `QString`. IMO it is not too different to `QStringList(Iter begin, Iter end)` where `Iter` is actually the iterator of an `OrderSet<Tag>` converted to `QString`.